### ISSUE TITLE
psalm cache bug with constructor property promotion

### ISSUE THREAD

**Issue by ging-dev** [vimeo/psalm#10080]
```php
<?php

class Example
{
    public function __construct(private string $error)
    {
    }

    public function other(): string
    {
        return $this->error;
    }
}
```

run `psalm` with cache. After, change content to:

```php
<?php

class Example
{
    public function __construct()
    {
    }

    public function other(): string
    {
        return $this->error;
    }
}
```

re-run `psalm`, `psalm` doesn't show an error message, but an error actually exists

**Comment by weirdan** [comment]
Can you provide a reproducer repo? I tried this in Psalm's own repo, and errors were reported.

**Comment by weirdan** [comment]
Relevant parts from the debug log:
```console
$ vendor/bin/psalm --debug 
Process plugin adjustments...
Target PHP version: 8.2 (inferred from current PHP version).
Scanning files...
1 changed files:
    /home/weirdan/src/psalm/issues/psalm-cache-reproduce/src/Bug/Reproduce.php
[.....]
Parsing /home/weirdan/src/psalm/issues/psalm-cache-reproduce/src/Bug/Reproduce.php
Deep scanning /home/weirdan/src/psalm/issues/psalm-cache-reproduce/src/Bug/Reproduce.php
[.....]
Analyzing files...
Getting /home/weirdan/src/psalm/issues/psalm-cache-reproduce/src/Bug/Reproduce.php
Analyzing /home/weirdan/src/psalm/issues/psalm-cache-reproduce/src/Bug/Reproduce.php
Skipping analysis of pre-analyzed method App\Bug\Reproduce::error
[.....]
```

**Comment by weirdan** [comment]
Changes to `__construct()` should cause `error()` to be reanalyzed, and it should be triggered somewhere around here: https://github.com/vimeo/psalm/blob/c83b0c96a9d832ec67084116ffb5e11710c9a131/src/Psalm/Internal/Codebase/Analyzer.php#L658-L663
